crates/interpreter/src/valid.rs

```diff
@@ -586,7 +586,9 @@ impl<'a, 'm> Expr<'a, 'm> {
             Block(b) => self.push_label(self.blocktype(&b)?, LabelKind::Block)?,
             Loop(b) => {
                 let type_ = self.blocktype(&b)?;
-                self.push_label(type_, LabelKind::Loop(self.branch_target(type_.params.len())))?
+                let mut target = self.branch_target(type_.params.len());
+                target.parser = offset_parser(target, -2);
```

Comment:
This only works with simple block types. The correct solution is to save the parser position before parsing the instruction line 561 and doing `target.parser = saved` here.

Reply:
I believe this would happen if there are more than 127 types in the module because then if a loop has parameters, then it has to use a type index (the `x:s33` in https://webassembly.github.io/spec/core/binary/instructions.html#control-instructions) which will need 2 bytes.